packages/app/components/Questions/Queue/JoinTagGroupButton.tsx

```diff
@@ -0,0 +1,220 @@
+
+  // just used to disable the button so students can't modify their question while being helped
+  const beingHelped =
+    (questionType &&
+      studentQuestion &&
+      studentQuestion.status === OpenQuestionStatus.Helping) ||
+    (taskId && studentDemo && studentDemo.status === OpenQuestionStatus.Helping)
+
+  const onClick = async (event) => {
+    event.stopPropagation()
+    setIsLoading(true)
+    const isJoining = !isJoined
+    try {
+      if (isJoining) {
+        if (questionType) {
+          if (!studentQuestion) {
+            // if student doesn't already have a question, create one
+            await createQuestion('', [questionType], false, false)
+          } else {
+            // if the student already has a question, update it with the new question type appended
+            const newQuestionTypes = [
+              ...studentQuestion.questionTypes,
+              questionType,
+            ]
+            await updateQuestion(
+              studentQuestion.text,
+              newQuestionTypes,
+              studentQuestion.groupable,
+              studentQuestion.isTaskQuestion,
+              studentQuestion.location,
+            )
+          }
+        } else if (taskId) {
+          // basically the same thing, but with tasks
+          if (!studentDemo) {
+            // tasks get put as "Mark "taskId" "taskId2" and have no questionTypes
+            await createQuestion(`Mark "${taskId}"`, [], false, true)
+          } else {
+            const newQuestionText = studentDemo.text + ` "${taskId}"`
+            await updateQuestion(
+              newQuestionText,
+              [],
+              studentDemo.groupable,
+              studentDemo.isTaskQuestion,
+              studentDemo.location,
+            )
+          }
+        } else {
+          console.log(
+            'Error: This component was used incorrectly. No questionType or taskId provided',
+          )
+        }
+      } else {
+        // If leaving the tag group, remove the questionType (for tag) or remove off the task from the question text
+        if (questionType) {
+          if (!studentQuestion) {
+            console.log('Error: Student does not have a question to leave')
+            return
+          }
+          if (
+            studentQuestion.text === '' &&
+            studentQuestion.questionTypes.length === 1
+          ) {
+            await leaveQueue(false)
+          } else {
+            const newQuestionTypes = studentQuestion.questionTypes.filter(
+              (qt) => qt.id !== questionType.id,
+            )
+            await updateQuestion(
+              studentQuestion.text,
+              newQuestionTypes,
+              studentQuestion.groupable,
+              studentQuestion.isTaskQuestion,
+              studentQuestion.location,
+            )
+          }
+        } else if (taskId) {
+          if (!studentDemo) {
+            console.log('Error: Student does not have a task to leave')
```

Comment:
Can we remvove any mentions of console log as it's mainly for development purposes

Reply:
right, i will replace them with console.error. I feel like its important to log any error states when using the component since that means something is wrong.